Civi/Core/SettingsMetadata.php

```diff
@@ -141,30 +139,54 @@ protected static function _filterSettingsSpecification($filters, &$settingSpec)
    * Retrieve options from settings metadata
    *
    * @param array $settingSpec
+   * @param bool|array $getOptions
    */
-  protected static function loadOptions(&$settingSpec) {
+  protected static function fillOptions(&$settingSpec, $getOptions) {
```

Comment:
So the thing I find kinda messy is that getOptions is really a 3 way option it

- NO
- YES
- do it as an array

I feel like is_array() to determine that last one is a bit opaque

Reply:
Well, yea but that's way outside the scope of this PR IMO.

Reply:
It's really an APIv4 documentation issue.

Reply:
@colemanw well maybe at the api level - but at the level of this function we could pass something pretty clear

Reply:
Unfortunately, I don't think we can. To maintain backward-compat it needs to be able to return a flat array if `true` and in order to correctly serve data to APIv4 it needs to be able to fetch name, label, etc if array.

Reply:
Re backward-compat - it's only called from one place in the code so the contract can change I think - I mean you DID rename it!

Reply:
in fact getOptions now should be renamed to $returnOptionsAsArray since it's a new param to the function

Reply:
Well the function is only called from one place but it returns values to `SettingsMetadata::getMetadata` which is used in legacy places that expect flat arrays of options.
Since it's protected I took the liberty of renaming it :)

Reply:
Sure I'll rename that param too.

Reply:
I renamed it but resisted the temptation to turn it into a boolean because eventually we'll add support for more return values.

Reply:
Hmm I feel like Civi is full of that sort of good intention code which makes less & less sense after the extra thing isn't done months later. I'd probably go with what actually makes sense with the code now as it's a protected function - but failing that beef up the comments explaining the rationale for the param in the docblock

Reply:
@eileenmcnaughton I renamed the variable & added a comment. I don't think this is one of those cases because it's not broken or half-functional. It works, this just leaves the door open for further enhancements.

Reply:
OK - it still feels a bit too much like magic to me but at least the docs cover it now